microjogos/2023S1/projeto-pochete/cenas/Head.gd

```diff
@@ -40,7 +43,11 @@ func check_move(pos, dir, delta):
 	area_ray.target_position = target_pos
 	ray.force_raycast_update()
 	area_ray.force_raycast_update()
+	var over_tail = (area_ray.is_colliding()
+					and area_ray.get_collider() in tail_nodes
+					and area_ray.get_collider() != tail_nodes[-1])
```

Comment:
Adiciona lógica para impedir andar sobre a cauda aqui por causa do novo passo maior.
Bloqueia o movimento se ele for andar sobre a cauda exceto se for a última parte da cauda, que significa que o jogador está voltando para trás.